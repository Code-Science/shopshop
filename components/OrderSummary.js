import React from 'react';
import styles from '../styles/components/CheckoutForm.module.css';
import { useStateValue } from '../store/StateProvider';

const OrderSummary = () => {
  const [{ basket }] = useStateValue();

  const totalPrice = () => {
    return basket.reduce((sum, curr) => {
      return sum + curr.price * curr.quantity;
    }, 0);
  };

  return (
    <div className={styles.orderSummary}>
      <div className={styles.divider}></div>
      <div>
        <h1 className={styles.orderSummaryHeading}>Order Summary</h1>
        <div className={styles.orderSummaryDetails}>
          <h3>Standard Delivery Charges</h3>
          <p>£10.00 per 5kg</p>
          <hr></hr>
          <h3>Today's charge: £{totalPrice().toFixed(2)}</h3>
        </div>
      </div>
    </div>
  );
};

export default OrderSummary;
