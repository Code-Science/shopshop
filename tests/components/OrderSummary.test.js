import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import OrderSummary from '../../components/OrderSummary';

jest.mock('../../store/StateProvider');

configure({ adapter: new Adapter() });

describe('Order Summary Component', () => {
  const wrapper = shallow(<OrderSummary />);

  it('expect to render OrderSummary component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
