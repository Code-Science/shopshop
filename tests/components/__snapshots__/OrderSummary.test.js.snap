// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Summary Component snapshot test 1`] = `
<div
  className="orderSummary"
>
  <div
    className="divider"
  />
  <div>
    <h1
      className="orderSummaryHeading"
    >
      Order Summary
    </h1>
    <div
      className="orderSummaryDetails"
    >
      <h3>
        Standard Delivery Charges
      </h3>
      <p>
        £10.00 per 5kg
      </p>
      <hr />
      <h3>
        Today's charge: £
        NaN
      </h3>
    </div>
  </div>
</div>
`;
